gcode/cmds.go

```diff
@@ -49,87 +72,108 @@ type RotateZ struct {
 	Angle float64
 }
 
-func (r RotateZ) ToGCode(b *bytes.Buffer) {
+func (r RotateZ) ToGCode(b *bytes.Buffer, st State, sett GcodeSettings) State {
 	b.WriteString("G0 A" + StrF(r.Angle) + "\n")
+	return st
+
 }
 func (r RotateZ) LayersCount() int {
 	return 0
 }
 
 type LayersMoving struct {
-	Layers    []Layer
-	Index     int
-	ExtParams ExtrusionParams
+	Layers []Layer
 }
 
-func (lm LayersMoving) ToGCode(b *bytes.Buffer) {
-	eOff := 0.0
-	//reset extruder encoder
-	b.WriteString("G92 E0\n")
```

Comment:
Why you removed reseting extruder and the beginning of gcode?

Reply:
header template has initial setting and i think it is not needed. Only for resetting if it is too long (somewhere below)